src/components/checkbox/GroupableCheckboxConnected.tsx

```diff
@@ -0,0 +1,44 @@
+import { connect } from 'react-redux';
+import * as _ from 'underscore';
+import { IReactVaporState } from '../../ReactVapor';
+import { IReduxAction, ReduxUtils } from '../../utils/ReduxUtils';
+import { IInputDispatchProps, IInputProps, IInputStateProps } from '../input/Input';
+import {Checkbox} from './Checkbox';
+import {ICheckboxState} from './CheckboxReducers';
+import {
+  addGroupedCheckbox, IGroupableCheckboxActionPayload, removeGroupedCheckbox,
+  toggleGroupedCheckbox,
+} from './GroupableCheckboxActions';
+import {IGroupableCheckboxesState} from './GroupableCheckboxReducers';
+
+export interface IGroupableCheckboxOwnProps extends IInputProps {
+  isParent?: boolean;
+  parentId?: string;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IGroupableCheckboxOwnProps): IInputStateProps => {
+  const groupableCheckboxesState: IGroupableCheckboxesState = _.findWhere(state.groupableCheckboxes, { parentId: ownProps.parentId || ownProps.id });
+  if (groupableCheckboxesState) {
+    const checkbox: ICheckboxState = ownProps.isParent ? groupableCheckboxesState.parent : _.findWhere(groupableCheckboxesState.checkboxes, {id: ownProps.id});
+    return {
+      checked: checkbox && checkbox.checked,
+      indeterminate: ownProps.isParent && groupableCheckboxesState.nbSelected > 0 &&
```

Comment:
Each condition on its own line starting with the operator first https://github.com/coveo/tsjs#control-statements